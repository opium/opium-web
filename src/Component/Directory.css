.ThumbnailList {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    align-items: stretch;
}

.Footer {
    text-align: right;
}

.DirectoryHeaderWithBanner {
    height: 400px;
    background-position: center center;
    background-size: cover;
    margin-bottom: 80px;
}

.DirectoryHeaderTitle {
    margin-top: 70px;
    color: #fff;
    text-align: center;
    font-size: 54px;
}

.Back {
    display: inline-block;
    padding: 1em;
    font-size: 1.2em;
    color: #899eaf;
    text-decoration: none;
}

.Back:hover {
    text-decoration: underline;
}
