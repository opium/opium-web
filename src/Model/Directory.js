import { List, Map, Record } from 'immutable';
import { entityFactory, mapEntityRelationShips } from './Factory';

class Directory extends Record({
  id: null,
  name: null,
  slug: null,
  children: [],
  _links: {},
  _embedded: {},
  imageLines: [],
  parent: null,
  directoryThumbnail: null,
}) {
  constructor(val) {
    const data = val;
    if (val.image_lines) {
      data.imageLines = List(val.image_lines.map(line =>
        Map(Object.keys(line).map(key => [key, line[key]]))
      ));
    }

     data.directoryThumbnail = val._embedded &&
       val._embedded.directory_thumbnail &&
       entityFactory(val._embedded.directory_thumbnail);

    return mapEntityRelationShips(super(data), data);
  }

  getChildById(id) {
    return this.children.find(item => item.id === parseInt(id, 10));
  }

  getChildrenSize() {
    return this.children.size || 0;
  }

  getPositionBounds() {
    let bounds = Map();
    const children = this.children.filter(c => !!c.position);

    if (children.size === 1) {
      const child = children.first().position;

      return Map({
        top: child.get('lat') + 1,
        bottom: child.get('lat') - 1,
        left: child.get('lng') - 1,
        right: child.get('lng') + 1,
      });
    }

    for (const child of children) {
      const latitude = child.position.get('lat');
      const longitude = child.position.get('lng');

      if (!bounds.get('top') || bounds.get('top') < latitude) {
        bounds = bounds.set('top', latitude);
      }
      if (!bounds.get('bottom') || bounds.get('bottom') > latitude) {
        bounds = bounds.set('bottom', latitude);
      }
      if (!bounds.get('right') || bounds.get('right') < longitude) {
        bounds = bounds.set('right', longitude);
      }
      if (!bounds.get('left') || bounds.get('left') > longitude) {
        bounds = bounds.set('left', longitude);
      }
    }

    if (!bounds.get('top')) {
      return null;
    }

    const latDelta = (bounds.get('top') - bounds.get('bottom')) / 4;
    const lngDelta = (bounds.get('right') - bounds.get('left')) / 4;

    return bounds.merge({
      top: bounds.get('top') + latDelta,
      bottom: bounds.get('bottom') - latDelta,
      left: bounds.get('left') - lngDelta,
      right: bounds.get('right') + lngDelta,
    });
  }

  // hack until https://github.com/mapado/rest-client-js-sdk/issues/11 is resolved
  get(key) {
    if (key === '@id') {
      return `/v1/directories/${this.id}`;
    }

    return super.get(key);
  }
}

export default Directory;
