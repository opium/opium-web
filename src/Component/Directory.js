import React, { Component, PropTypes } from 'react';
import { Link } from 'react-router';
import Helmet from 'react-helmet';
import cn from 'classnames';
import ChevronLeft from 'react-icons/lib/ti/chevron-left';
import './Directory.css';
import Thumbnail from './Thumbnail';
import File from '../Model/File';
import ImageWithLoader from './ImageWithLoader';

class DirectoryHeader extends Component {
  static propTypes = {
    directory: PropTypes.object.isRequired,
    viewportWidth: PropTypes.number.isRequired,
  };

  constructor(props) {
    super(props);

    this.state = {
      backgroundLoaded: null,
    };
  }

  componentDidUpdate(prevProps) {
    if (prevProps.directory !== this.props.directory) {
      this.setState({
        backgroundLoaded: null,
      });
    }
  }

  render() {
    const { directory, viewportWidth } = this.props;

    if (!directory.parent && !directory.directoryThumbnail) {
      return null;
    }


    const styles = {};
    if (directory.directoryThumbnail) {
      if (this.state.backgroundLoaded) {
        const backgroundImage = directory.directoryThumbnail.generateCrop(viewportWidth, 400);
        styles.backgroundImage = `linear-gradient(rgba(0, 0, 0, 0) 50%,
          rgba(0, 0, 0, 0.8)),
          url(${backgroundImage})`;
      }
    }

    return (
      <header
        className={cn(
          directory.directoryThumbnail && 'DirectoryHeader--WithBanner',

        )}
      >
        {directory.directoryThumbnail &&
          <div>
            <ImageWithLoader
              src={directory.directoryThumbnail.generateCrop(viewportWidth, 400)}
              onLoad={() => this.setState({ backgroundLoaded: true })}
              loaderProps={{ className: 'DirectoryHeader__Image' }}
            />
            <div
              className={cn(
                'DirectoryHeader__Image',
                this.state.backgroundLoaded && 'DirectoryHeader__Image--Opaque'
              )}
              style={styles}
            />
          </div>
        }

        {directory.parent &&
          <Link to={`/${directory.parent.slug}`} className="DirectoryHeader__Back">
            <ChevronLeft />
            Back to albums
          </Link>
        }

        {directory.directoryThumbnail &&
          <h1 className="DirectoryHeader__Title">
            {directory.name}
          </h1>
        }
      </header>
    );
  }
};

class Directory extends Component {
  static propTypes = {
    directory: PropTypes.object,
    findDirectory: PropTypes.func.isRequired,
    slug: PropTypes.string.isRequired,
    viewportWidth: PropTypes.number.isRequired,
  }

  constructor(props) {
    super(props);

    this.renderOneLine = this.renderOneLine.bind(this);
  }

  componentDidMount() {
    this.props.findDirectory(this.props.slug);
  }

  componentDidUpdate(prevProps) {
    if (this.props.slug !== prevProps.slug) {
      this.props.findDirectory(this.props.slug);
    }
  }

  renderOneLine(line) {
    const directory = this.props.directory;

    return line.entrySeq().map(([id, thumbnail]) => {
      const child = directory.getChildById(id);
      const childSlug = child instanceof File ?
        `/${directory.slug}/${child.slug}` :
        `/${child.slug}`;

      return (
        <Link to={childSlug}>
          <Thumbnail
            key={id}
            title={child.name}
            image={thumbnail.thumbs}
          />
        </Link>
      );
    });
  }

  render() {
    if (this.props.children) {
      return this.props.children;
    }

    const directory = this.props.directory;

    if (!directory) {
      return <div />;
    }

    return (
      <div>
        <Helmet title={directory.name} />

        <DirectoryHeader
          directory={directory}
          viewportWidth={this.props.viewportWidth}
        />

        <div className="Directory__ThumbnailList">
          {directory.imageLines.map(this.renderOneLine)}
        </div>

        <footer className="DirectoryFooter">
          Total number of items: {directory.getChildrenSize()}
        </footer>
      </div>
    );
  }
}

export default Directory;
