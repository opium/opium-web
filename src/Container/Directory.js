import { connect } from 'react-redux';
import { Set } from 'immutable';
import Directory from '../Component/Directory';
import { find } from '../Action/DirectoryAction';
import { loadImage } from '../Action/ImageAction';

function mapStateToProps(state, ownProps) {
  const viewportWidth = document.body.clientWidth;
  const loadedImages = state.imageLoader.get('loadedImages') || Set();
  const directory = state.opium.get('currentDirectory');
  const hasBackground = !!(directory && directory.directoryThumbnail);
  const directoryImage = hasBackground && directory.directoryThumbnail.generateCrop(viewportWidth, 400);

  return {
    directory,
    slug: ownProps.params.directorySlug || '',
    hasBackground,
    backgroundLoaded: directory && loadedImages.has(directoryImage),
    backgroundImage: directoryImage,
  }
};

const mapDispatchToProps = {
  findDirectory: find,
  loadImage,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Directory);
