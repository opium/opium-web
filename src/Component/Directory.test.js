jest.mock('react-leaflet');

import React from 'react';
import renderer from 'react-test-renderer';
import Directory from './Directory';
import DirectoryModel from '../Model/Directory';

it('renders directory without crashing', () => {
  const component = renderer.create(<Directory
    findDirectory={jest.fn()}
    slug=""
    loadImage={jest.fn()}
    hasBackground={false}
    displayAdminLink
  />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot()
});

it('renders an empty directory', () => {
  const directory = new DirectoryModel({
    name: 'foo',
  });
  const component = renderer.create(<Directory
    directory={directory}
    findDirectory={jest.fn()}
    slug=""
    loadImage={jest.fn()}
    hasBackground={false}
    displayAdminLink
  />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot()
});

it('renders an empty directory without admin link', () => {
  const directory = new DirectoryModel({
    name: 'foo',
  });
  const component = renderer.create(<Directory
    directory={directory}
    findDirectory={jest.fn()}
    slug=""
    loadImage={jest.fn()}
    hasBackground={false}
    displayAdminLink={false}
  />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot()
});

it('renders a non empty directory', () => {
  const directory = new DirectoryModel({
    name: 'foo',
    children: [
      {
        id: 1,
        name: "Alps",
        slug: "alps",
        type: 'directory',
      },
      {
        id: 2,
        name: "Pacific Ocean",
        slug: "pacific-ocean",
        type: 'directory',
      }
    ],
    image_lines: [
      {
        "1": {
          thumbs: 'http://demo.opium.sitioweb.fr/2909-vallon-moy-res-jpg/thumbs/200-200',
        },
        "2": {
          thumbs: 'http://demo.opium.sitioweb.fr/2010-mavericks-competition-edit1-jpg/thumbs/200-200',
        }
      }
    ],
  });
  const component = renderer.create(<Directory
    directory={directory}
    findDirectory={jest.fn()}
    slug=""
    loadImage={jest.fn()}
    hasBackground={false}
    displayAdminLink
  />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot()
});

it('renders a non empty directory with parent', () => {
  const directory = new DirectoryModel({
    name: 'foo',
    children: [
      {
        id: 1,
        name: "Alps",
        slug: "alps",
        type: 'directory',
      },
      {
        id: 2,
        name: "Pacific Ocean",
        slug: "pacific-ocean",
        type: 'directory',
      }
    ],
    image_lines: [
      {
        "1": {
          thumbs: 'http://demo.opium.sitioweb.fr/2909-vallon-moy-res-jpg/thumbs/200-200',
        },
        "2": {
          thumbs: 'http://demo.opium.sitioweb.fr/2010-mavericks-competition-edit1-jpg/thumbs/200-200',
        }
      }
    ],
    parent: {
      name: 'bar',
      slug: 'bar',
    },
  });
  const component = renderer.create(<Directory
    directory={directory}
    findDirectory={jest.fn()}
    slug=""
    loadImage={jest.fn()}
    hasBackground={false}
    displayAdminLink
  />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot()
});

it('renders a directory with header', () => {
  const directory = new DirectoryModel({
    name: 'foo',
    children: [
      {
        id: 1,
        name: "Alps",
        slug: "alps",
        type: 'directory',
      },
      {
        id: 2,
        name: "Pacific Ocean",
        slug: "pacific-ocean",
        type: 'directory',
      }
    ],
    image_lines: [
      {
        "1": {
          thumbs: 'http://demo.opium.sitioweb.fr/2909-vallon-moy-res-jpg/thumbs/200-200',
        },
        "2": {
          thumbs: 'http://demo.opium.sitioweb.fr/2010-mavericks-competition-edit1-jpg/thumbs/200-200',
        }
      }
    ],
    parent: {
      name: 'bar',
      slug: 'bar',
    },
    _embedded: {
      directory_thumbnail: {
        type: 'file',
        _links: {
          cropTemplate: {
            href: 'http://demo.opium.sitioweb.fr/2010-mavericks-competition-edit1-jpg/thumbs/%7BcropWidth%7D-%7BcropHeight%7D',
          }
        },
      }
    }
  });
  const component = renderer.create(<Directory
    directory={directory}
    findDirectory={jest.fn()}
    slug=""
    loadImage={jest.fn()}
    hasBackground={true}
    backgroundLoaded={false}
    backgroundImage="http://demo.opium.sitioweb.fr/2909-vallon-moy-res-jpg/thumbs/1170-400"
    displayAdminLink
  />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot()
});

it('renders a directory with children and loaded background', () => {
  const directory = new DirectoryModel({
    name: 'foo',
    children: [
      {
        id: 1,
        name: "Alps",
        slug: "alps",
        type: 'directory',
      },
      {
        id: 2,
        name: "Pacific Ocean",
        slug: "pacific-ocean",
        type: 'directory',
      }
    ],
    image_lines: [
      {
        "1": {
          thumbs: 'http://demo.opium.sitioweb.fr/2909-vallon-moy-res-jpg/thumbs/200-200',
        },
        "2": {
          geometry: { width: 200, height: 200 }
        }
      }
    ],
    parent: {
      name: 'bar',
      slug: 'bar',
    },
    _embedded: {
      directory_thumbnail: {
        type: 'file',
        _links: {
          cropTemplate: {
            href: 'http://demo.opium.sitioweb.fr/2010-mavericks-competition-edit1-jpg/thumbs/%7BcropWidth%7D-%7BcropHeight%7D',
          }
        },
      }
    }
  });
  const component = renderer.create(<Directory
    directory={directory}
    findDirectory={jest.fn()}
    slug=""
    loadImage={jest.fn()}
    hasBackground={true}
    backgroundLoaded={true}
    backgroundImage="http://demo.opium.sitioweb.fr/2909-vallon-moy-res-jpg/thumbs/1170-400"
    displayAdminLink
  />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot()
});
