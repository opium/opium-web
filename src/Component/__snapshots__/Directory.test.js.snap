exports[`test renders a directory with children and loaded background 1`] = `
<div>
  <header
    className="DirectoryHeader DirectoryHeader--WithBanner">
    <div>
      <div
        className="Loader DirectoryHeader__Image">
        <div
          className="Loader__BallScaleRippleMultiple">
          <div
            style={
              Object {
                "borderColor": "#1eb694",
              }
            } />
          <div
            style={
              Object {
                "borderColor": "#1eb694",
              }
            } />
          <div
            style={
              Object {
                "borderColor": "#1eb694",
              }
            } />
        </div>
      </div>
      <div
        className="DirectoryHeader__Image DirectoryHeader__Image--Opaque"
        style={
          Object {
            "backgroundImage": "linear-gradient(rgba(0, 0, 0, 0) 50%,
                  rgba(0, 0, 0, 0.8)),
                  url(http://demo.opium.sitioweb.fr/2909-vallon-moy-res-jpg/thumbs/1170-400)",
          }
        } />
    </div>
    <a
      className="DirectoryHeader__Back"
      onClick={[Function]}
      style={Object {}}>
      <svg
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": undefined,
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width="1em">
        <g>
          <path
            d="m24 9.3c-1.3-1.3-3.4-1.3-4.7 0l-10.7 10.7 10.7 10.7c0.7 0.6 1.5 1 2.4 1s1.7-0.4 2.3-1c1.3-1.3 1.3-3.4 0-4.7l-5.9-6 5.9-6c1.3-1.3 1.3-3.4 0-4.7z" />
        </g>
      </svg>
      Back to albums
    </a>
    <a
      className="DirectoryHeader__UploadLink"
      onClick={[Function]}
      style={Object {}}>
      <svg
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": undefined,
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width="1em">
        <g>
          <path
            d="m35 26.7c0-0.2 0-0.4-0.1-0.6l-3.3-10c-0.2-0.6-0.9-1.1-1.6-1.1h-6.7v3.3h5.5l2.8 8.4h-23.2l2.8-8.4h5.5v-3.3h-6.7c-0.7 0-1.4 0.5-1.6 1.1l-3.3 10c-0.1 0.2-0.1 0.4-0.1 0.6 0 0 0 8.3 0 8.3 0 0.9 0.7 1.7 1.7 1.7h26.6c1 0 1.7-0.8 1.7-1.7 0 0 0-8.3 0-8.3z m-8.3-13.5c0.4 0 0.8-0.2 1.1-0.5 0.7-0.6 0.7-1.7 0-2.4l-7.8-7.8-7.8 7.8c-0.7 0.7-0.7 1.8 0 2.4 0.3 0.3 0.7 0.5 1.1 0.5s0.9-0.2 1.2-0.5l3.8-3.8v11.1c0 0.9 0.8 1.7 1.7 1.7s1.7-0.8 1.7-1.7v-11.1l3.8 3.8c0.3 0.3 0.7 0.5 1.2 0.5z" />
        </g>
      </svg>
      Upload file
    </a>
    <h1
      className="DirectoryHeader__Title">
      foo
    </h1>
  </header>
  <div
    className="Directory__ThumbnailList">
    <a
      onClick={[Function]}
      style={Object {}}>
      <article
        className="Thumbnail"
        style={
          Object {
            "backgroundImage": "url(http://demo.opium.sitioweb.fr/2909-vallon-moy-res-jpg/thumbs/200-200)",
          }
        }>
        <span
          className="Thumbnail_Caption">
          Alps
        </span>
      </article>
    </a>
    <a
      onClick={[Function]}
      style={Object {}}>
      <article
        className="Thumbnail"
        style={Object {}}>
        <svg
          className="Thumbnail_FolderPlaceholder"
          fill="currentColor"
          height={200}
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width={200}>
          <g>
            <path
              d="m30 10h-10c0-1.8-1.5-3.3-3.3-3.3h-6.7c-2.8 0-5 2.2-5 5v16.6c0 2.8 2.2 5 5 5h20c2.8 0 5-2.2 5-5v-13.3c0-2.8-2.2-5-5-5z m-20 0h6.7c0 1.8 1.5 3.3 3.3 3.3h10c0.9 0 1.7 0.8 1.7 1.7h-23.4v-3.3c0-1 0.8-1.7 1.7-1.7z m20 20h-20c-0.9 0-1.7-0.7-1.7-1.7v-11.6h23.4v11.6c0 1-0.8 1.7-1.7 1.7z" />
          </g>
        </svg>
        <span
          className="Thumbnail_Caption">
          Pacific Ocean
        </span>
      </article>
    </a>
  </div>
  <footer
    className="DirectoryFooter">
    Total number of items: 
    2
  </footer>
</div>
`;

exports[`test renders a directory with header 1`] = `
<div>
  <header
    className="DirectoryHeader DirectoryHeader--WithBanner">
    <div>
      <div
        className="Loader DirectoryHeader__Image">
        <div
          className="Loader__BallScaleRippleMultiple">
          <div
            style={
              Object {
                "borderColor": "#1eb694",
              }
            } />
          <div
            style={
              Object {
                "borderColor": "#1eb694",
              }
            } />
          <div
            style={
              Object {
                "borderColor": "#1eb694",
              }
            } />
        </div>
      </div>
      <div
        className="DirectoryHeader__Image"
        style={Object {}} />
    </div>
    <a
      className="DirectoryHeader__Back"
      onClick={[Function]}
      style={Object {}}>
      <svg
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": undefined,
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width="1em">
        <g>
          <path
            d="m24 9.3c-1.3-1.3-3.4-1.3-4.7 0l-10.7 10.7 10.7 10.7c0.7 0.6 1.5 1 2.4 1s1.7-0.4 2.3-1c1.3-1.3 1.3-3.4 0-4.7l-5.9-6 5.9-6c1.3-1.3 1.3-3.4 0-4.7z" />
        </g>
      </svg>
      Back to albums
    </a>
    <a
      className="DirectoryHeader__UploadLink"
      onClick={[Function]}
      style={Object {}}>
      <svg
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": undefined,
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width="1em">
        <g>
          <path
            d="m35 26.7c0-0.2 0-0.4-0.1-0.6l-3.3-10c-0.2-0.6-0.9-1.1-1.6-1.1h-6.7v3.3h5.5l2.8 8.4h-23.2l2.8-8.4h5.5v-3.3h-6.7c-0.7 0-1.4 0.5-1.6 1.1l-3.3 10c-0.1 0.2-0.1 0.4-0.1 0.6 0 0 0 8.3 0 8.3 0 0.9 0.7 1.7 1.7 1.7h26.6c1 0 1.7-0.8 1.7-1.7 0 0 0-8.3 0-8.3z m-8.3-13.5c0.4 0 0.8-0.2 1.1-0.5 0.7-0.6 0.7-1.7 0-2.4l-7.8-7.8-7.8 7.8c-0.7 0.7-0.7 1.8 0 2.4 0.3 0.3 0.7 0.5 1.1 0.5s0.9-0.2 1.2-0.5l3.8-3.8v11.1c0 0.9 0.8 1.7 1.7 1.7s1.7-0.8 1.7-1.7v-11.1l3.8 3.8c0.3 0.3 0.7 0.5 1.2 0.5z" />
        </g>
      </svg>
      Upload file
    </a>
    <h1
      className="DirectoryHeader__Title">
      foo
    </h1>
  </header>
  <div
    className="Directory__ThumbnailList">
    <a
      onClick={[Function]}
      style={Object {}}>
      <article
        className="Thumbnail"
        style={
          Object {
            "backgroundImage": "url(http://demo.opium.sitioweb.fr/2909-vallon-moy-res-jpg/thumbs/200-200)",
          }
        }>
        <span
          className="Thumbnail_Caption">
          Alps
        </span>
      </article>
    </a>
    <a
      onClick={[Function]}
      style={Object {}}>
      <article
        className="Thumbnail"
        style={
          Object {
            "backgroundImage": "url(http://demo.opium.sitioweb.fr/2010-mavericks-competition-edit1-jpg/thumbs/200-200)",
          }
        }>
        <span
          className="Thumbnail_Caption">
          Pacific Ocean
        </span>
      </article>
    </a>
  </div>
  <footer
    className="DirectoryFooter">
    Total number of items: 
    2
  </footer>
</div>
`;

exports[`test renders a non empty directory 1`] = `
<div>
  <header
    className="DirectoryHeader">
    <a
      className="DirectoryHeader__UploadLink"
      onClick={[Function]}
      style={Object {}}>
      <svg
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": undefined,
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width="1em">
        <g>
          <path
            d="m35 26.7c0-0.2 0-0.4-0.1-0.6l-3.3-10c-0.2-0.6-0.9-1.1-1.6-1.1h-6.7v3.3h5.5l2.8 8.4h-23.2l2.8-8.4h5.5v-3.3h-6.7c-0.7 0-1.4 0.5-1.6 1.1l-3.3 10c-0.1 0.2-0.1 0.4-0.1 0.6 0 0 0 8.3 0 8.3 0 0.9 0.7 1.7 1.7 1.7h26.6c1 0 1.7-0.8 1.7-1.7 0 0 0-8.3 0-8.3z m-8.3-13.5c0.4 0 0.8-0.2 1.1-0.5 0.7-0.6 0.7-1.7 0-2.4l-7.8-7.8-7.8 7.8c-0.7 0.7-0.7 1.8 0 2.4 0.3 0.3 0.7 0.5 1.1 0.5s0.9-0.2 1.2-0.5l3.8-3.8v11.1c0 0.9 0.8 1.7 1.7 1.7s1.7-0.8 1.7-1.7v-11.1l3.8 3.8c0.3 0.3 0.7 0.5 1.2 0.5z" />
        </g>
      </svg>
      Upload file
    </a>
  </header>
  <div
    className="Directory__ThumbnailList">
    <a
      onClick={[Function]}
      style={Object {}}>
      <article
        className="Thumbnail"
        style={
          Object {
            "backgroundImage": "url(http://demo.opium.sitioweb.fr/2909-vallon-moy-res-jpg/thumbs/200-200)",
          }
        }>
        <span
          className="Thumbnail_Caption">
          Alps
        </span>
      </article>
    </a>
    <a
      onClick={[Function]}
      style={Object {}}>
      <article
        className="Thumbnail"
        style={
          Object {
            "backgroundImage": "url(http://demo.opium.sitioweb.fr/2010-mavericks-competition-edit1-jpg/thumbs/200-200)",
          }
        }>
        <span
          className="Thumbnail_Caption">
          Pacific Ocean
        </span>
      </article>
    </a>
  </div>
  <footer
    className="DirectoryFooter">
    Total number of items: 
    2
  </footer>
</div>
`;

exports[`test renders a non empty directory with parent 1`] = `
<div>
  <header
    className="DirectoryHeader">
    <a
      className="DirectoryHeader__Back"
      onClick={[Function]}
      style={Object {}}>
      <svg
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": undefined,
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width="1em">
        <g>
          <path
            d="m24 9.3c-1.3-1.3-3.4-1.3-4.7 0l-10.7 10.7 10.7 10.7c0.7 0.6 1.5 1 2.4 1s1.7-0.4 2.3-1c1.3-1.3 1.3-3.4 0-4.7l-5.9-6 5.9-6c1.3-1.3 1.3-3.4 0-4.7z" />
        </g>
      </svg>
      Back to albums
    </a>
    <a
      className="DirectoryHeader__UploadLink"
      onClick={[Function]}
      style={Object {}}>
      <svg
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": undefined,
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width="1em">
        <g>
          <path
            d="m35 26.7c0-0.2 0-0.4-0.1-0.6l-3.3-10c-0.2-0.6-0.9-1.1-1.6-1.1h-6.7v3.3h5.5l2.8 8.4h-23.2l2.8-8.4h5.5v-3.3h-6.7c-0.7 0-1.4 0.5-1.6 1.1l-3.3 10c-0.1 0.2-0.1 0.4-0.1 0.6 0 0 0 8.3 0 8.3 0 0.9 0.7 1.7 1.7 1.7h26.6c1 0 1.7-0.8 1.7-1.7 0 0 0-8.3 0-8.3z m-8.3-13.5c0.4 0 0.8-0.2 1.1-0.5 0.7-0.6 0.7-1.7 0-2.4l-7.8-7.8-7.8 7.8c-0.7 0.7-0.7 1.8 0 2.4 0.3 0.3 0.7 0.5 1.1 0.5s0.9-0.2 1.2-0.5l3.8-3.8v11.1c0 0.9 0.8 1.7 1.7 1.7s1.7-0.8 1.7-1.7v-11.1l3.8 3.8c0.3 0.3 0.7 0.5 1.2 0.5z" />
        </g>
      </svg>
      Upload file
    </a>
  </header>
  <div
    className="Directory__ThumbnailList">
    <a
      onClick={[Function]}
      style={Object {}}>
      <article
        className="Thumbnail"
        style={
          Object {
            "backgroundImage": "url(http://demo.opium.sitioweb.fr/2909-vallon-moy-res-jpg/thumbs/200-200)",
          }
        }>
        <span
          className="Thumbnail_Caption">
          Alps
        </span>
      </article>
    </a>
    <a
      onClick={[Function]}
      style={Object {}}>
      <article
        className="Thumbnail"
        style={
          Object {
            "backgroundImage": "url(http://demo.opium.sitioweb.fr/2010-mavericks-competition-edit1-jpg/thumbs/200-200)",
          }
        }>
        <span
          className="Thumbnail_Caption">
          Pacific Ocean
        </span>
      </article>
    </a>
  </div>
  <footer
    className="DirectoryFooter">
    Total number of items: 
    2
  </footer>
</div>
`;

exports[`test renders an empty directory 1`] = `
<div>
  <header
    className="DirectoryHeader">
    <a
      className="DirectoryHeader__UploadLink"
      onClick={[Function]}
      style={Object {}}>
      <svg
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": undefined,
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width="1em">
        <g>
          <path
            d="m35 26.7c0-0.2 0-0.4-0.1-0.6l-3.3-10c-0.2-0.6-0.9-1.1-1.6-1.1h-6.7v3.3h5.5l2.8 8.4h-23.2l2.8-8.4h5.5v-3.3h-6.7c-0.7 0-1.4 0.5-1.6 1.1l-3.3 10c-0.1 0.2-0.1 0.4-0.1 0.6 0 0 0 8.3 0 8.3 0 0.9 0.7 1.7 1.7 1.7h26.6c1 0 1.7-0.8 1.7-1.7 0 0 0-8.3 0-8.3z m-8.3-13.5c0.4 0 0.8-0.2 1.1-0.5 0.7-0.6 0.7-1.7 0-2.4l-7.8-7.8-7.8 7.8c-0.7 0.7-0.7 1.8 0 2.4 0.3 0.3 0.7 0.5 1.1 0.5s0.9-0.2 1.2-0.5l3.8-3.8v11.1c0 0.9 0.8 1.7 1.7 1.7s1.7-0.8 1.7-1.7v-11.1l3.8 3.8c0.3 0.3 0.7 0.5 1.2 0.5z" />
        </g>
      </svg>
      Upload file
    </a>
  </header>
  <div
    className="Directory__ThumbnailList" />
  <footer
    className="DirectoryFooter">
    Total number of items: 
    0
  </footer>
</div>
`;

exports[`test renders an empty directory without admin link 1`] = `
<div>
  <header
    className="DirectoryHeader" />
  <div
    className="Directory__ThumbnailList" />
  <footer
    className="DirectoryFooter">
    Total number of items: 
    0
  </footer>
</div>
`;

exports[`test renders directory without crashing 1`] = `<div />`;
